### PR TITLE
Chaishi/aggregator

### PR DESCRIPTION
**Pull request by changchaishi**
This pull request consists of
- news api update ( but deprecated )
- merge with incoming dockerize feature ( from cswang/Docker )
- addition of aggregator field on user schema
- powerdata api update ( introducing aggregator features, and including some conflict fix )
- update api reference ( from chaishi/api_reference branch ( but closed ), known issues fixed, add GET /user is_aggregator output)
- fix db is_aggregator nullity problem in user model

### CODE REVIEW

File: pt/endpoints/powerdata/resource.py [hunk 1]
```diff
@@ -58,43 +58,51 @@ def _set_get_parser(self):
...
     def get(self):
         args = self.get_parser.parse_args()
-
-        logger.info(f"[Get Datas Request]\nUser Account:{g.account}\nUUID:{g.uuid}\n")
-
+        logger.info(f"[Get Datas Request]\nUser Account:{g.account}\nUUID:{g.uuid}\nIs_Aggregator:{g.is_aggregator}\n")
+        if g.is_aggregator is True and args["participant_id"]:
+            user_id = args["participant_id"]
+        else:
+            user_id = g.uuid
```
Comment (ttw225) [inline pt/endpoints/powerdata/resource.py:77]: how about "ALL" user's data?
概覽功能

File: pt/endpoints/powerdata/resource.py [hunk 2]
```diff
@@ -58,43 +58,51 @@ def _set_get_parser(self):
...
             end_time = datetime.strptime(args["end_time"], "%Y/%m/%d")
             # if start time same as end time, default use three days' data
             if start_time == end_time:
-                start_time = start_time - timedelta(days=2)
-            end_time += timedelta(days=1)
-            return self.chart_mode(start_time, end_time)
-
+                start_time = str(
+                    datetime.strptime(start_time, "%Y/%m/%d").date() - timedelta(days=2)
+                )
```
Comment (ttw225) [inline pt/endpoints/powerdata/resource.py]: Should not be `string` type. Have you test it?

File: apiary.apib [hunk 3]
```diff
@@ -130,54 +144,53 @@ curl http://140.116.247.120:5000/news -X GET -H "Content-Type: application/json"
...
+curl http://140.116.247.120:5000/bidsubmit?per_page=10&page=5&bid_type=buy -X GET -H "Content-Type: application/json" -H "Authorization: Bearer 3MaTIcta709SxWZ88OkaLjKvNzgfFkxqr8WemUjeOKLZcImscV6WcziuFyfrbXjc"
 </pre>
 
-Return the bid content.
+Return the bid content valid from the day (Today, current time).
 
-+ Request (application/json)
++ Parameters
+    + per_page: `10` (number) - 一頁的標案數量.
+    + page: `5` (number) - 頁面書.
```
Comment (ttw225) [inline apiary.apib]: Did you mean `頁面數`？

File: apiary.apib [hunk 4]
```diff
@@ -375,84 +379,139 @@ NOT YET !!!!!!
...
++ field: `NCKU` (string) - 場域
++ cluster: `1` (number) - 風機編號
++ WindGridPower: `0.000` (number) - 風機功率
++ updated_at: `2019-09-04T12:00:00` - 資料上傳時間
+
+### PowerEntry1
++ address: `https://thetangle.org/transaction/WGQEE9ZNHHNPZ9AILVK9NAUXTRQZWJVEMIXWPBMWFQZTRXLIURDKLVUKACJAQIJOCKDWAZTLTLZY99999` - IOTA address
++ data_type: `ESS` - 產能設備
++ date: `2020/02/14` - 日期
++ id：`049a9c28-c0cc-418e-9f35-23f6bc43df7e` - PowerData UUID
```
Comment (ttw225) [inline apiary.apib]: Watch out! Should use `:` instead of `：` !!！!!

File: apiary.apib [hunk 5]
```diff
@@ -375,84 +379,139 @@ NOT YET !!!!!!
...
++ power_display: `4.771` (number) - 電力資訊
++ time: `23:50` - 時間
+
+### PowerEntry2
++ address: `https://thetangle.org/transaction/QAYPUZOKFXCDXGL9ZUZNVMKRNYGPSYRDQNJDQBGL9HYZRHBZ9FPJFWXRIGTXJSKMADGPODWYMCQYZ9999` - IOTA address
++ data_type: `WT` - 產能設備
++ date: `2020/02/14` - 日期
++ id：`049a9c28-c0cc-418e-9f35-23f6bc43df7e` - PowerData UUID
```
Comment (ttw225) [inline apiary.apib]: Watch out! Should use `:` instead of `：` !!！!!

File: apiary.apib [hunk 6]
```diff
@@ -130,54 +144,53 @@ curl http://140.116.247.120:5000/news -X GET -H "Content-Type: application/json"
...
-            + content: Website Upload to Version 0.2 - The Message Content
+            + time: `2019/08/25 10:00` - The Message Upload Date
+            + content: `Website Upload to Version 0.2` - The Message Content
 
 ## Bid Submit [/bidsubmit]
 
 Related to bidding.
 
-### Bidding Project [GET]
+### Bidding Project [GET /power_info{?per_page}{?page}{?bid_type}]
```
Comment (ttw225) [inline apiary.apib]: Must be `/bidsubmit`

File: apiary.apib [hunk 7]
```diff
@@ -321,48 +309,64 @@ curl http://140.116.247.120:5000/participant -X GET -H "Content-Type: applicatio
...
-curl http://140.116.247.120:5000/power_info -X GET -H "Content-Type: application/json" -H "Authorization: Bearer OXBiogjCfdKb7m9CEOPhpbjjv0n5OtfltzgMJbSxbP7jVFY8XUNzHqkGidDumQSR"
-curl http://140.116.247.120:5000/power_info?time=2019-10-19 -X GET -H "Content-Type: application/json" -H "Authorization: Bearer OXBiogjCfdKb7m9CEOPhpbjjv0n5OtfltzgMJbSxbP7jVFY8XUNzHqkGidDumQSR"
+curl http://140.116.247.120:5000/power_info?per_page=10&page=1&time=2020/02/14 -X GET -H "Content-Type: application/json" -H "Authorization: Bearer OXBiogjCfdKb7m9CEOPhpbjjv0n5OtfltzgMJbSxbP7jVFY8XUNzHqkGidDumQSR"
 </pre>
 
 + Parameters
-    + time: `2019-10-19` (optional) - Query date. Format: `YYYY-mm-DD`
-        + Default: System date
+    + per_page: `10` (number, optional) - Power data per page.
+    + page: `1` (number, optional) - Power data at page.
```
Comment (ttw225) [inline apiary.apib]: If there's no setting of `per_page` or `page`, return data will be empty.
For `PowerData Data Table Mode`, these parameters are both required.

File: apiary.apib [hunk 8]
```diff
@@ -321,48 +309,64 @@ curl http://140.116.247.120:5000/participant -X GET -H "Content-Type: applicatio
...
+        + totalCount: `785` (number) - Query 時間內資料總數
 
+### PowerData (Data Chart Mode) [GET /power_info{?start_time}{?end_time}]
+<pre>
+curl http://140.116.247.120:5000/power_info?start_time=2020/02/14?end_time=2020/02/15 -X GET -H "Content-Type: application/json" -H "Authorization: Bearer OXBiogjCfdKb7m9CEOPhpbjjv0n5OtfltzgMJbSxbP7jVFY8XUNzHqkGidDumQSR"
+</pre>
+
++ Parameters
+    + start_time: `2020/02/14` (string, optional) - Date for the data, format YYYY/MM/DD.
+    + end_time: `2020/02/15` (string, optional) - Date for the data, format YYYY/MM/DD.
```
Comment (ttw225) [inline apiary.apib]: These two parameters must be required to `PowerData Chart Mode`

File: apiary.apib [hunk 9]
```diff
@@ -375,84 +379,139 @@ NOT YET !!!!!!
...
+    + price: `50` (number) - Winning Total Price
++ upload: `Thu, 12 Mar 2020 14:15:00 GMT` - Upload Time
++ achievement: `100%` - 0~100%
++ settlement: `50` (number) - Settlement Transaction Price
+
+### BidSubmit1
++ bid_type: `buy` (string) - 投標類型
++ date: `2020/10/29` (string) - 目標執行日期 YYYY/MM/DD
++ end_time: `2020/10/29 19` (string) - 執行結束時間
++ id：`049a9c28-c0cc-418e-9f35-23f6bc43df7e` - 投標 UUID
```
Comment (ttw225) [inline apiary.apib]: Watch out! Should use : instead of ： !!！!!

File: apiary.apib [hunk 10]
```diff
@@ -375,84 +379,139 @@ NOT YET !!!!!!
...
++ time: `18` (number) - 執行時間
++ total_price: `68.0` (number) - 總交易價格
++ upload_time: `Sat, 21 Mar 2020 14:28:59 GMT` (string) - 投標上傳時間
++ volume: `10` (number) - 投標電量
+
+### BidSubmit2
++ bid_type: `buy` (string) - 投標類型
++ date: `2020/10/230` (string) - 目標執行日期 YYYY/MM/DD
++ end_time: `2020/10/30 7` (string) - 執行結束時間
++ id：`b6953af5-0803-4559-af5c-115613627624` - 投標 UUID
```
Comment (ttw225) [inline apiary.apib]: Watch out! Should use : instead of ： !!！!!

**Review by ttw225 — COMMENTED** [review]
Need to fix some problems~

File: pt/endpoints/powerdata/resource.py (hunk 2 above)
Comment (ttw225) [reply pt/endpoints/powerdata/resource.py]: related to the newest base_model

File: pt/endpoints/powerdata/resource.py (hunk 1 above)
Comment (changchaishi) [reply pt/endpoints/powerdata/resource.py:77]: I thought we decided to cancel out the overview for powerdata for aggregotor, and make the aggregator's powerdata as default. Perhaps we can mark it as an upcoming issue first.

File: pt/endpoints/powerdata/resource.py (hunk 2 above)
Comment (changchaishi) [reply pt/endpoints/powerdata/resource.py]: Noted.
Logic itself is wrong. Should be corrected as `start_time -= timedelta(days=2)`

File: apiary.apib (hunk 3 above)
Comment (changchaishi) [reply apiary.apib]: Noted

File: apiary.apib (hunk 4 above)
Comment (changchaishi) [reply apiary.apib]: Noted.
Will be replace by 'find and replace'

File: apiary.apib (hunk 5 above)
Comment (changchaishi) [reply apiary.apib]: Noted.
Will be replace by 'find and replace'

File: apiary.apib (hunk 6 above)
Comment (changchaishi) [reply apiary.apib]: Noted.

File: apiary.apib (hunk 7 above)
Comment (changchaishi) [reply apiary.apib]: Noted.

File: apiary.apib (hunk 8 above)
Comment (changchaishi) [reply apiary.apib]: Noted.

File: apiary.apib (hunk 7 above)
Comment (iknowright) [reply apiary.apib:334]: Noted.
Will be replace by 'find and replace'

File: apiary.apib (hunk 10 above)
Comment (changchaishi) [reply apiary.apib]: Noted.
Will be replace by 'find and replace'

File: apiary.apib (hunk 9 above)
Comment (changchaishi) [reply apiary.apib]: Noted.
Will be replace by 'find and replace'

File: apiary.apib (hunk 4 above)
Comment (ttw225) [reply apiary.apib]: add a space between `:` and `

File: apiary.apib (hunk 5 above)
Comment (ttw225) [reply apiary.apib]: add a space between `:` and `

File: apiary.apib (hunk 9 above)
Comment (ttw225) [reply apiary.apib]: add a space between `:` and `

File: apiary.apib (hunk 10 above)
Comment (ttw225) [reply apiary.apib]: add a space between `:` and `

**Review by ttw225 — APPROVED** [review]
Add some missing spaces and merge this PR.

**Review by ttw225 — APPROVED** [review]
LGTM

**Pull request merged**